### PR TITLE
[DEV-728] Setup GitHub workflow and update documentation

### PR DESCRIPTION
**Pull request by RashikaKarki**
This PR contains updates in documentation and adds setup of GitHub actions to run test. Also, added Rubocop.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  - Go to the repository page
+  - Click on the "Releases" tab
+  - Click on the "Draft a new release" button
+  - Fill the release information, including the tag version, release title, and release notes
+- Build and publish the Gem
+```
+gem build omniauth-mlh.gemspec
+gem push omniauth-mlh-X.Y.Z.gem
+```
+Replace `X.Y.Z` with the new version number you want to release.
```
Comment (erinosher) [inline CONTRIBUTING.md]: This should probably specify that we follow [Semver](https://semver.org/)

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+- Create your feature branch (`git checkout -b my-new-feature`)
+- Commit your changes (`git commit -am 'Add some feature'`)
+- Push to the branch (`git push origin my-new-feature`)
+- Create a new Pull Request
+
+## Local Testing Strategy
+
+To test omniauth-mlh locally, you can use [Sinatra application](https://gist.github.com/theycallmeswift/d8a8a22f95dd5d35f03661b19665e0d3). Here are the steps to set up and run the application:
+
+- Add files from [Sinatra application](https://gist.github.com/theycallmeswift/d8a8a22f95dd5d35f03661b19665e0d3) in a folder within a directory next to the checked out [omniauth-mlh](https://github.com/MLH/omniauth-mlh) library.
```
Comment (erinosher) [inline CONTRIBUTING.md]: I'm inclined to say this should probably live in an `example` directory. But happy to defer to @theycallmeswift's take

File: README.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
-# MLH/my-mlh-omniauth
+# MLH/omniauth-mlh
 
-[![Build Status](https://travis-ci.org/MLH/my-mlh-omniauth.svg?branch=master)](https://travis-ci.org/MLH/my-mlh-omniauth)
+[![Build Status](https://travis-ci.org/MLH/omniauth-mlh.svg?branch=master)](https://travis-ci.org/MLH/omniauth-mlh)
```
Comment (erinosher) [inline README.md]:
```suggestion
[![Test](https://github.com/MLH/omniauth-mlh/actions/workflows/test.yml/badge.svg)](https://github.com/MLH/omniauth-mlh/actions/workflows/test.yml)
```

This should use the GitHub Actions badge not Travis CI

**Review by erinosher — COMMENTED** [review]
Left a couple of small Notes, but otherwise looks good to me.

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+jobs:
+  test:
+    runs-on: ubuntu-latest 
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: '2.7'
```
Comment (theycallmeswift) [inline .github/workflows/test.yml]: Should we test this on a few different versions of Ruby? Specifically 2.7 and the latest 3.x make sense to me

File: CONTRIBUTING.md (hunk 1 above)
Comment (theycallmeswift) [reply CONTRIBUTING.md]: We should remove this section anyway. It's not on the contributor to maintain the version number, that's on the maintainer (us). I do think we may want to document the release process somewhere though. Maybe in the Engineering gitbook?

Also, this is missing the tagging steps:

```
$ git commit -m "vX.Y.Z"
$ git tag -a vX.Y.Z -m "vX.Y.Z"
$ git push origin main --tags
```

File: CONTRIBUTING.md (hunk 2 above)
Comment (theycallmeswift) [reply CONTRIBUTING.md]: We could also generalize this the way this Gem does: https://github.com/auth0/omniauth-auth0/blob/master/CONTRIBUTING.md

Agree that we should avoid linking to a random gist

File: omniauth-mlh.gemspec [hunk 5]
```diff
@@ -8,8 +8,8 @@ require 'omniauth-mlh/version'
 Gem::Specification.new do |spec|
   spec.name          = 'omniauth-mlh'
   spec.version       = OmniAuth::MLH::VERSION
-  spec.authors       = ['Swift']
-  spec.email         = ['swift@mlh.io']
+  spec.authors       = ['MLH']
```
Comment (theycallmeswift) [inline omniauth-mlh.gemspec]: 'Major League Hacking (MLH)'

**Review by theycallmeswift — COMMENTED** [review]
We should also add some kind of rubocop linting to ensure code quality / formatting. Otherwise LGTM aside from comments!

File: CONTRIBUTING.md (hunk 1 above)
Comment (RashikaKarki) [reply CONTRIBUTING.md]: I agree. For now I will take this out and keep a backup with tagging step for the Engineering gitbook or any other alternative knowledge base.

File: .github/workflows/test.yml (hunk 4 above)
Comment (RashikaKarki) [reply .github/workflows/test.yml]: Sounds good to me. I will add another version too.

File: .rubocop.yml [hunk 6]
```diff
@@ -1,12 +1,34 @@
 AllCops:
-  TargetRubyVersion: '2.2.0'
+  TargetRubyVersion: '2.7.0'
```
Comment (theycallmeswift) [inline .rubocop.yml:7]: Can you add this comment above this line?

```
# Ruby 2.7 is the MINIMUM Ruby version supported

File: .rubocop.yml [hunk 7]
```diff
@@ -1,12 +1,34 @@
...
 
+# Naming section allows you to enforce consistent naming conventions across your codebase.
 Naming/FileName:
   Exclude:
     - 'lib/omniauth-mlh.rb'
 
+# Metrics section allows you to enforce code complexity rules to keep the codebase maintainable.
 Metrics/BlockLength:
```
Comment (theycallmeswift) [inline .rubocop.yml]: I prefer:

```
Metrics/BlockLength:
  CountComments: true 
  Max: 25
  Exclude:
    - "spec/**/*"
```

File: .rubocop.yml [hunk 8]
```diff
@@ -1,12 +1,34 @@
...
   Exclude:
     - 'Rakefile'
     - '**/*.rake'
     - 'spec/**/*'
+
+# Two spaces, no tabs (for indentation).
+Layout/IndentationWidth:
```
Comment (theycallmeswift) [inline .rubocop.yml]: This is enabled by default: https://github.com/rubocop/rubocop/blob/master/config/default.yml

File: .rubocop.yml [hunk 9]
```diff
@@ -1,12 +1,34 @@
...
+  Enabled: true
+
+# No trailing whitespace.
+Layout/TrailingWhitespace:
```
Comment (theycallmeswift) [inline .rubocop.yml]: This is enabled by default: https://github.com/rubocop/rubocop/blob/master/config/default.yml#L1514

File: .rubocop.yml [hunk 10]
```diff
@@ -1,12 +1,34 @@
...
+  Enabled: true
+
+# Blank lines should not have any spaces.
+Layout/TrailingEmptyLines:
```
Comment (theycallmeswift) [inline .rubocop.yml]: This is enabled by default: https://github.com/rubocop/rubocop/blob/master/config/default.yml#L1503

File: .rubocop.yml [hunk 11]
```diff
@@ -1,12 +1,34 @@
...
+  Enabled: true
+
+# Use Ruby >= 1.9 syntax for hashes. Prefer { a: :b } over { :a => :b }.
+Style/HashSyntax:
```
Comment (theycallmeswift) [inline .rubocop.yml]: This is enabled by default: https://github.com/rubocop/rubocop/blob/master/config/default.yml#L3912

File: .rubocop.yml [hunk 12]
```diff
@@ -1,12 +1,34 @@
...
+  Enabled: true
+
+# Use my_method(my_arg) not my_method( my_arg ) or my_method my_arg.
+Lint/RequireParentheses:
```
Comment (theycallmeswift) [inline .rubocop.yml]: This is enabled by default: https://github.com/rubocop/rubocop/blob/master/config/default.yml#L2195

File: .rubocop.yml [hunk 13]
```diff
@@ -1,12 +1,34 @@
 AllCops:
-  TargetRubyVersion: '2.2.0'
+  TargetRubyVersion: '2.7.0'
 
+# Naming section allows you to enforce consistent naming conventions across your codebase.
```
Comment (theycallmeswift) [inline .rubocop.yml]: Make this comment specific to the reason you're overriding the default behavior, not a description of the default. E.g. Answer the question **why** are you excluding `lib/omniauth-mlh.rb`?

File: Rakefile [hunk 14]
```diff
@@ -1,8 +1,13 @@
 #!/usr/bin/env rake
 require 'bundler/gem_tasks'
 require 'rspec/core/rake_task'
+require 'rubocop/rake_task'
 
 RSpec::Core::RakeTask.new
+RuboCop::RakeTask.new do |task|
```
Comment (theycallmeswift) [inline Rakefile]: You don't need the block here at all. Rubocop will pick up the `.rubocop.yml` file by default and will include all lintable files. You're actually missing some files with your patterns definiton.

File: Rakefile [hunk 15]
```diff
@@ -1,8 +1,13 @@
...
+    task.patterns = ['**/*.rb']
+    task.options = ['-c', '.rubocop.yml']
+end
 
 desc 'Run specs'
 task default: :spec
```
Comment (theycallmeswift) [inline Rakefile:12]: `task default: [:rubocop, :spec]`

File: .rubocop.yml [hunk 16]
```diff
@@ -1,12 +1,34 @@
 AllCops:
```
Comment (theycallmeswift) [inline .rubocop.yml:5]: I think you'll address a lot of this in the ticket for adding Rubocop to OHQ, but here are some good resources:

- https://github.com/thoughtbot/guides/blob/main/ruby/.rubocop.yml
- https://github.com/rubocop/rubocop/blob/master/config/default.yml

**Review by theycallmeswift — CHANGES_REQUESTED** [review]
Some more minor feedback, mostly related to rubocop

File: .rubocop.yml [hunk 17]
```diff
@@ -1,12 +1,12 @@
...
   Exclude:
-    - 'lib/omniauth-mlh.rb'
+    - vendor/**/*
 
-Metrics/BlockLength:
-  Exclude:
-    - 'Rakefile'
-    - '**/*.rake'
-    - 'spec/**/*'
+RSpec/MultipleExpectations:
```
Comment (theycallmeswift) [inline .rubocop.yml]: I actually dislike this rule. I think we should disable it. This is a good default to include in OHQ as well!

```
RSpec/MultipleExpectations:
  Description: Allow tests to contain multiple expectations
  Enabled: false
```

File: Rakefile [hunk 18]
```diff
@@ -1,8 +1,12 @@
...
 require 'bundler/gem_tasks'
 require 'rspec/core/rake_task'
+require 'rubocop/rake_task'
 
 RSpec::Core::RakeTask.new
+RuboCop::RakeTask.new
 
 desc 'Run specs'
-task default: :spec
+task default: %i[spec rubocop]
```
Comment (theycallmeswift) [inline Rakefile]: This is the thing we are disabling in OHQ. We should disable it here as well

**Review by theycallmeswift — APPROVED** [review]
Overall this is about ready to merge. I would include the new "default" rules you're writing for the OHQ Rubocop config (using array notation instead of `%i`, etc) and I also found a good Rspec rule that we can disable here and also include in our new OHQ config!

File: .rubocop.yml [hunk 19]
```diff
@@ -1,12 +1,23 @@
...
-    - 'lib/omniauth-mlh.rb'
+    - vendor/**/*
 
-Metrics/BlockLength:
-  Exclude:
-    - 'Rakefile'
-    - '**/*.rake'
-    - 'spec/**/*'
+Style/WordArray:
+  EnforcedStyle: brackets
```
Comment (theycallmeswift) [inline .rubocop.yml:13]: You're missing the `Description` and `Enabled` attributes for clarity.

File: .rubocop.yml [hunk 20]
```diff
@@ -1,12 +1,23 @@
...
+
+Style/SymbolArray: 
+  Description: Force symbol arrays to use bracket notation 
+  EnforcedStyle: brackets 
+  Enabled: true
+
+Style/RegexpLiteral:
+  AllowInnerSlashes: true
```
Comment (theycallmeswift) [inline .rubocop.yml:23]: You're missing the Description and Enabled attributes for clarity.

File: .rubocop.yml (hunk 17 above)
Comment (theycallmeswift) [reply .rubocop.yml]: This was never resolved

File: Rakefile (hunk 18 above)
Comment (theycallmeswift) [reply Rakefile]: This isn't resolved

File: spec/support/shared_examples.rb [hunk 21]
```diff
@@ -5,35 +5,40 @@
```
Comment (theycallmeswift) [inline spec/support/shared_examples.rb:5]: This file still has `%i[ ]` bracket notation

**Review by theycallmeswift — CHANGES_REQUESTED** [review]

**Review by theycallmeswift — APPROVED** [review]
LGTM!

**Pull request merged**